src/EditorFeatures/CSharpTest/SimplifyInterpolation/SimplifyInterpolationTests.cs

```diff
@@ -0,0 +1,534 @@
+    {
+        _ = $""prefix {someValue{|Unnecessary:[||].PadLeft(|}3{|Unnecessary:, ' ')|}} suffix"";
+    }
+}",
+@"class C
+{
+    void M(string someValue)
+    {
+        _ = $""prefix {someValue,-3} suffix"";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task PadRightWithSpaceChar()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(string someValue)
+    {
+        _ = $""prefix {someValue{|Unnecessary:[||].PadRight(|}3{|Unnecessary:, ' ')|}} suffix"";
+    }
+}",
+@"class C
+{
+    void M(string someValue)
+    {
+        _ = $""prefix {someValue,3} suffix"";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task PadLeftWithNonSpaceChar()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(string someValue)
+    {
+        _ = $""prefix {someValue[||].PadLeft(3, '\t')} suffix"";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task PadRightWithNonSpaceChar()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(string someValue)
+    {
+        _ = $""prefix {someValue[||].PadRight(3, '\t')} suffix"";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task PadRightWithComplexConstantExpression()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(string someValue)
+    {
+        const int someConstant = 1;
+        _ = $""prefix {someValue{|Unnecessary:[||].PadRight(|}(byte)3.3 + someConstant{|Unnecessary:)|}} suffix"";
+    }
+}",
+@"class C
+{
+    void M(string someValue)
+    {
+        const int someConstant = 1;
+        _ = $""prefix {someValue,(byte)3.3 + someConstant} suffix"";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ToStringWithNoParameterWhenFormattingComponentIsSpecified()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(string someValue)
+    {
+        _ = $""prefix {someValue[||].ToString():goo} suffix"";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ToStringWithStringLiteralParameterWhenFormattingComponentIsSpecified()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    void M(string someValue)
+    {
+        _ = $""prefix {someValue[||].ToString(""bar""):goo} suffix"";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ToStringWithNoParameterWhenAlignmentComponentIsSpecified()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M(string someValue)
+    {
+        _ = $""prefix {someValue{|Unnecessary:[||].ToString()|},3} suffix"";
+    }
+}",
+@"class C
+{
+    void M(string someValue)
+    {
+        _ = $""prefix {someValue,3} suffix"";
+    }
+}");
+        }
+
+        [Fact]
+        public async Task ToStringWithNoParameterWhenBothComponentsAreSpecified()
+        {
+            await TestMissingInRegularAndScriptAsync(
```

Comment:
@ryzngard I wrote the test this way because I thought of it as a correctness issue. The `goo` format string is being applied to the return value of `ToString()` and is a no-op. Removing ToString() would cause it to stop being a no-op. Wouldn't it be better to put a warning squiggle on `:goo` than to mark ToString() as redundant?

Reply:
That would be a separate feature though, a warning squiggle whenever it is known that the format component cannot be applied to the expression type.